Microsoft.Toolkit.Uwp.UI.Controls/ColorPickerButton/ColorPickerButton.Properties.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.ObjectModel;
+using Windows.UI.Xaml;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// Contains all properties for the <see cref="ColorPickerButton"/>.
+    /// </summary>
+    public partial class ColorPickerButton
+    {
+        /// <summary>
+        /// Identifies the <see cref="CustomPaletteColors"/> dependency property.
+        /// </summary>
+        public static readonly DependencyProperty CustomPaletteColorsProperty =
```

Comment:
I'm trying to understand the distinction here between the properties on the control and the interface of `IColorPalette`? Shouldn't these go together?

Reply:
The fundamental difference is an IColorPalette is 2-dimensional. CustomPaletteColors is 1-dimensions with an additional property to define when to wrap to the next row. IColorPalette is designed to work in several other cases. It's columns contain unique colors and it's rows contain shades of the color -- generally speaking. This means you can get a specific color index and then automatically vary the shade as desired. That's a much higher level concept than the ColorPickerButton supports though. However, other controls in the future -- such as pie charts -- could use the same palette interface to change the entire shade of the pie chart while keeping colors the same.